### PR TITLE
Add ReadOnly property to TextBox component

### PR DESCRIPTION
**Pull request by matthewcoufal**
This PR adds a property `ReadOnly` to the `TextBox` component. This addresses issue https://github.com/mit-cml/appinventor-sources/issues/1841.

### CODE REVIEW

**Comment by ewpatton** [comment]
@matthewcoufal Are you sure that this has the correct behavior? I thought the original issue is that setting Enabled = false doesn't allow text to be selected in the textview, and on the surface this appears to me to not do anything different. Am I missing something?

File: appinventor/appengine/src/com/google/appinventor/client/youngandroid/YoungAndroidFormUpgrader.java [hunk 1]
```diff
@@ -1517,6 +1517,10 @@ private static int upgradeTextBoxProperties(Map<String, JSONValue> componentProp
       // RequestFocus method was added
       srcCompVersion = 5;
     }
+    if (scrCompVersion < 6) {
```
Comment (SusanRatiLane) [inline appinventor/appengine/src/com/google/appinventor/client/youngandroid/YoungAndroidFormUpgrader.java]: Compile failed. This is a typo that should be srcCompVersion.

**Comment by AppInventorWorkerBee** [comment]
Can one of the admins verify this patch?

File: appinventor/appengine/src/com/google/appinventor/client/youngandroid/YoungAndroidFormUpgrader.java (hunk 1 above)
Comment (matthewcoufal) [reply appinventor/appengine/src/com/google/appinventor/client/youngandroid/YoungAndroidFormUpgrader.java]: @SusanRatiLane just fixed that. Thanks for pointing this out!

File: appinventor/components/src/com/google/appinventor/components/runtime/TextBox.java [hunk 2]
```diff
@@ -183,6 +187,24 @@ public void MultiLine(boolean multiLine) {
     view.setSingleLine(!multiLine);
   }
 
+  @SimpleProperty(
+    category = PropertyCategory.BEHAVIOR,
+    description = "Whether the %type% is read-only. By default, this is true."
```
Comment (SusanRatiLane) [inline appinventor/components/src/com/google/appinventor/components/runtime/TextBox.java]: Don't you mean this property is false by default?

Comment (matthewcoufal) [reply appinventor/components/src/com/google/appinventor/components/runtime/TextBox.java]: @SusanRatiLane Yes! I will update that. Thanks!

**Comment by SusanRatiLane** [comment]
@AppInventorWorkerBee OK to test

**Pull request merged**